.github/workflows/workflow.yml

```diff
@@ -1,20 +1,161 @@
-  workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+  ## No sense in a dev building every time they push and no one
+  ## should be working on the common branches.
+  push:
+    branches:
+      - master
+      - develop
+      - 'hotfix/**'
+      - 'release/**'
+      - 'feature/**'
+  ## Any pull request. Yes the syntax looks weird
+  pull_request:
+
 jobs:
 
-  build:
 
+  test:
+    name: Test the loader on ${{matrix.operating-system}}
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+      - name: Get the code
+        uses: actions/checkout@v2
+
+      - name: Set up node
+        uses:  actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install packages
+        run:  |
+          npm ci
+
+      - name: Run tests
+        run:  |
+          npm test
+        env:
+          CI: true
+
+      - name: Archive test artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: test-results
+          path: coverage
+
+
+  integration_tests:
+    name: Run Integration Tests (Linux)
     runs-on: ubuntu-latest
+    needs: test
+
+    services:
+      elasticsearch:
+        image: elasticsearch:7.17.5
+        env:
+            discovery.type: single-node
+            ES_JAVA_OPTS: -Xms750m -Xmx750m
+        ports:
+          ## NOTE: This will be exposed as a random port referenced below by job.services.elasticsearch.ports[9200]
+          - 9200/tcp
+        options: --health-cmd="curl http://localhost:9200/_cluster/health" --health-interval=10s --health-timeout=5s --health-retries=10
 
     steps:
-      # Hello
-      - name: Run a one-line script
+
+      - name: Get the code
+        uses: actions/checkout@v2
+
+      - name: Set up node
+        uses:  actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install packages
+        run:  |
+          npm ci
+
+      - name: Run the loader to put data in Elasticsearch.
+        # It would be awesome if we could set up the config override in the job's first step
+        # in order to make it more obvious it was there and may need to be adjusted for
+        # whatever you're doing. Unfortunately, GitHub's magic for passing environment
+        # variables via $GITHUB_ENV doesn't seem to handle newlines very well, and this
+        # one is sufficiently complex, the newlines are vital to readability.
+        run:  |
+          export NODE_CONFIG="
+            {
+              \"pipeline\": {
+                  \"source\": {
+                      \"module\": \"./lib/sources/github-resource-source\",
+                      \"config\": {
+                          \"repoUrl\": \"https://github.com/nciocpl/r4r-loader\",
+                          \"branchName\": \"master\",
```

Comment:
Can you please pin this to an integration-branch so it will ALWAYS match the json.

Reply:
Done